code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilled_item
+	var/datum/looping_sound/grill/grill_loop
```

Comment:
If you wanna be real fancy, you could make this static. Not needed though

Reply:
how would that work

Reply:
Basically, you put a static in front of this and a ``= new`` at the end and then you the following in Initialize
```
grill_loop.start(src)
```
and in Destroy
```
grill_loop.stop(src)
```

Reply:
Actually, don't do that in Init and Destroy, just keep the code that is already here. Should work.

Reply:
If its static and you start it it'll start all of them though

Reply:
Yeah no... shit don't work like I thought it did.